### ISSUE TITLE
Unclosed file handlers lead to OSError when used in wsgi

### ISSUE THREAD

**Issue by ricardogsilva** [geopython/pywps#70]
I am getting

```python
OSError: [Errno 24] Too many open files
```

After leaving my pywps server running for a few days. I am running pywps under wsgi and calling it ~1500 times per day.

After investigating, it seems that everytime that pywps calls `tempfile.mkstemp` an open file handler is left lying around unclosed[1].

According to [python's documentation](https://docs.python.org/2.7/library/tempfile.html#tempfile.mkstemp), `mkstemp` returns a tuple with an already opened file and the path to the file. In pywps we are never using the file handlers (only the file paths) and we never close these open files!

In a wsgi application this becomes a serious problem, because wsgi does some caching of running processes. 

When using apache's mod_wsgi, the process is eventually restarted, causing all file descriptors to be closed. However, if your usage happens to be intensive (as seems to be my use case) there is a chance that you'll run out of file handlers to use.

I'll provide a Pull Request for solving this in the coming days.

[1] - To reproduce the problem, and assuming you get pywps running under wsgi, try the following:

```bash
# reload apache in order to start fresh
sudo service reload apache2

# extract the pid of the wsgi process
ps aux | grep [w]sgi

# check that there are no tmp* files opened by the wsgi process
sudo lsof -p <wsgi_process_pid> | grep tmp

# now run one of your WPS processes

# recheck if there are any tmp* files opened by the wsgi process
sudo lsof -p <wsgi_process_pid> | grep tmp

# there should now be one more tmp* file still open
# everytime you re-run a WPS process another tmp* file will be opened
# until sometime when apache's mod_wsgi decides to clean its cache
```

**Comment by tomkralidis** [comment]
Thanks @ricardogsilva!  FYI your PR is merged in master and backported to branch pywps-3.2.

**Comment by tomkralidis** [comment]
@gkm4d @ricardogsilva fix applied in 7e053dd78bde964557881cb7ea5db74525044af5
